beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -197,20 +217,18 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
         )
     }
 
-    pub fn put_gossip_data_columns(
+    pub fn put_gossip_data_column(
         &self,
-        slot: Slot,
-        block_root: Hash256,
-        gossip_data_columns: Vec<GossipVerifiedDataColumn<T>>,
+        gossip_data_column: GossipVerifiedDataColumn<T>,
     ) -> Result<Availability<T::EthSpec>, AvailabilityCheckError> {
-        let epoch = slot.epoch(T::EthSpec::slots_per_epoch());
-        let custody_columns = gossip_data_columns
-            .into_iter()
-            .map(|c| KzgVerifiedCustodyDataColumn::from_asserted_custody(c.into_inner()))
-            .collect::<Vec<_>>();
+        let custody_column = gossip_data_column.into_inner();
+
+        // Will never error as there's exactly one column
+        let custody_columns_same_block = KzgVerifiedDataColumnsSameBlock::new(vec![custody_column])
+            .expect("exactly one column is always in same block");
```

Comment:
I think it'd be better to make a new infallible constructor in `KzgVerifiedDataColumnsSameBlock` that takes a single column as a param